### PR TITLE
H-197: Add Linear connection OAuth API routes

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

Adds API routes to authorize Linear for use with HASH, and a basic frontend page at `/integrations/linear` for trying it out.

## 🔍 What does this change?

- Adds API routes `/oauth/linear` and `/oauth/linear/callback`
- Adds a FE testing page at `/integrations/linear`
- Adds supporting system types for `UserSecret` 
- Adds some of the clients given to the GraphQL resovlers as context to the request, so they're available in REST controllers
- 

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are in a state where docs changes are not _yet_ required but will be

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

File: libs/@local/hash-isomorphic-utils/src/ontology-types.ts [hunk 1]
```diff
@@ -243,6 +248,18 @@ const systemPropertyTypes = {
     title: "Deleted At",
     description: "Stringified timestamp of when something was deleted.",
   },
+  expiredAt: {
+    title: "Revoked At",
+    description: "Stringified timestamp of when something was revoked.",
+  },
```
Comment (benwerner01) [inline libs/@local/hash-isomorphic-utils/src/ontology-types.ts:254]:
```suggestion
  expiredAt: {
    title: "Expired At",
    description: "Stringified timestamp of when something was revoked.",
  },
```

File: apps/hash/README.md [hunk 2]
```diff
@@ -128,6 +128,16 @@ To run HASH locally, please follow these steps:
...
+1.  To develop using external integrations, you need a locally running Hashicorp Vault for secret storage
+
+    First, [install Vault](https://developer.hashicorp.com/vault/tutorials/getting-started/getting-started-install).
+
+    Then run a dev server:
+
+    ```sh
+    vault server -dev -dev-root-token-id=dev_root_token
+    ```
+
```
Comment (benwerner01) [inline apps/hash/README.md:140]: Not sure we need this as it's part of external services

File: apps/hash-api/src/integrations/linear/oauth.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    const ownerType =
+      extractEntityUuidFromEntityId(userEntityId) === ownedById
+        ? "user"
+        : "org";
+
+    const state = crypto.randomBytes(16).toString("hex");
+
+    stateMap.set(state, {
+      actorEntityId: req.user.entity.metadata.recordId.entityId,
+      expires: new Date(Date.now() + 1000 * 60 * 5),
```
Comment (TimDiekmann) [inline apps/hash-api/src/integrations/linear/oauth.ts:110]: Can we add a comment how long this will be? Is it 5 minutes?

Comment (CiaranMn) [reply apps/hash-api/src/integrations/linear/oauth.ts]: Done in https://github.com/hashintel/hash/pull/2796/commits/c9d1bbc67ed7aac3bec813803ebe7d27b3bf110f

File: apps/hash/README.md (hunk 2 above)
Comment (CiaranMn) [reply apps/hash/README.md]: Removed in https://github.com/hashintel/hash/pull/2796/commits/43e2b3ede58ec9f3fe936208a90bf2037c6d3c50

**Review by TimDiekmann — APPROVED** [review]

**Review by benwerner01 — APPROVED** [review]

**Pull request merged**